gnomad/utils/liftover.py

```diff
@@ -13,193 +13,148 @@
+    :param alleles: Input alleles.
+    :param destination_reference: Desired reference genome build for liftover.
+    :return: Struct containing expressions for lifted over locus/alleles as well as original locus/alleles.
+    """
+    lifted_over_locus = hl.liftover(locus, destination_reference, include_strand=True)
+    lifted_over_alleles = alleles.map(
+        lambda a: hl.if_else(
+            lifted_over_locus.is_negative_strand, hl.reverse_complement(a), a
         )
+    )
 
-    source.add_liftover(chain, target)
-    return [source, target]
+    return hl.struct(
+        new_locus=lifted_over_locus.result,
+        new_alleles=lifted_over_alleles,
+        original_locus=locus,
+        original_alleles=alleles,
+        locus_fail_liftover=hl.is_missing(lifted_over_locus),
+        ref_allele_mismatch=lifted_over_locus.sequence_context()
+        != lifted_over_alleles[0],
+    )
 
 
-def lift_data(
+def default_lift_data(
     t: Union[hl.MatrixTable, hl.Table],
-    gnomad: bool,
-    data_type: str,
-    path: str,
-    rg: hl.genetics.ReferenceGenome,
-    overwrite: bool,
 ) -> Union[hl.MatrixTable, hl.Table]:
     """
-    Lifts input Table or MatrixTable from one reference build to another
-
-    :param t: Table or MatrixTable
-    :param gnomad: Whether data is gnomAD data
-    :param data_type: Data type (exomes or genomes for gnomAD; not used otherwise)
-    :param path: Path to input Table/MatrixTable (if data is not gnomAD data)
-    :param rg: Reference genome
-    :param overwrite: Whether to overwrite data
-    :return: Table or MatrixTablewith liftover annotations
+    Lifts input Table or MatrixTable from one reference build to another.
+
+    :param t: Table or MatrixTable.
+    :return: Table or MatrixTable with liftover annotations.
     """
+    logger.info("Inferring input reference and destination reference...")
+    _, target_genome = get_liftover_genome(t)
 
-    logger.info("Annotating input with liftover coordinates")
-    liftover_expr = {
-        "new_locus": hl.liftover(t.locus, rg, include_strand=True),
-        "old_locus": t.locus,
-    }
+    logger.info("Annotating input data with liftover coordinates...")
     t = (
-        t.annotate(**liftover_expr)
+        t.annotate(**liftover_expr(t.locus, t.alleles, target_genome))
         if isinstance(t, hl.Table)
-        else t.annotate_rows(**liftover_expr)
+        else t.annotate_rows(**liftover_expr(t.locus, t.alleles, target_genome))
     )
 
-    no_target_expr = hl.agg.count_where(hl.is_missing(t.new_locus))
+    no_target_expr = hl.agg.count_where(t.locus_fail_liftover)
     num_no_target = (
         t.aggregate(no_target_expr)
         if isinstance(t, hl.Table)
         else t.aggregate_rows(no_target_expr)
     )
 
-    logger.info(f"Filtering out {num_no_target} sites that failed to liftover")
-    keep_expr = hl.is_defined(t.new_locus)
+    logger.info(f"Filtering out {num_no_target} sites that failed to liftover...")
```

Comment:
What are your thoughts on making the filtering, defaulting True, optional? The seqr loading pipeline annotates with liftover coordinates but does not filter. I'm not sure how many use cases there may be across other groups using these methods but there is one within ours(which is easy enough to implement by itself if you don't think it is needed).

Reply:
I like this idea. I think the default function probably should filter but like the option of specifying. will add